### PR TITLE
codec: simplify json codec.

### PR DESCRIPTION
**Pull request by hicqu**

### CODE REVIEW

**Comment by hanfei1991** [comment]
LGTM

File: util/types/json/json_test.go [hunk 1]
```diff
@@ -54,15 +54,16 @@ func (s *testJSONSuite) TestSerializeAndDeserialize(c *C) {
 	j2 := mustParseFromString(`[{"a": 1, "b": true}, 3, 3.5, "hello, world", null, true]`)
 
 	var testcses = []struct {
```
Comment (XuHuaiyu) [inline util/types/json/json_test.go]: s/testcses/testcases

File: util/types/json/serdes.go [hunk 2]
```diff
@@ -93,6 +93,43 @@ import (
...
+const (
+	typeCodeLen      int = 1
+	compoundCountLen     = 4
+	compoundSizeLen      = 4
+)
+
+// PeekBytesAsJSON trys to peek some bytes from b, until
+// we can deserialize a JSON from those bytes.
+func PeekBytesAsJSON(b []byte) (n int, err error) {
+	if len(b) > 0 {
```
Comment (XuHuaiyu) [inline util/types/json/serdes.go]: use len(b) <= 0 better?
thus avoiding the indentation

File: util/types/json/serdes.go [hunk 3]
```diff
@@ -93,6 +93,43 @@ import (
                              // lengths up to 16383, and so on...
 */
 
+const (
```
Comment (XuHuaiyu) [inline util/types/json/serdes.go]: add comments for these constants.

File: util/types/json/serdes.go [hunk 4]
```diff
@@ -93,6 +93,43 @@ import (
...
+	typeCodeLen      int = 1
+	compoundCountLen     = 4
+	compoundSizeLen      = 4
+)
+
+// PeekBytesAsJSON trys to peek some bytes from b, until
+// we can deserialize a JSON from those bytes.
+func PeekBytesAsJSON(b []byte) (n int, err error) {
+	if len(b) > 0 {
+		switch c := b[0]; c {
```
Comment (XuHuaiyu) [inline util/types/json/serdes.go]: Is TypeCode(b[0]) available?
If so, may be more readable.

File: util/types/json/serdes.go [hunk 5]
```diff
@@ -93,6 +93,43 @@ import (
...
+			if err == nil {
+				n = int(size) + int(reader.Size()) - int(reader.Len()) + typeCodeLen
+				return
+			}
+		case byte(typeCodeInt64), byte(typeCodeFloat64), byte(typeCodeLiteral):
+			n = jsonTypeCodeLength[TypeCode(c)] + typeCodeLen
+			return
+		}
+	}
+	err = errors.New("Invalid JSON bytes")
```
Comment (hhkbp2) [inline util/types/json/serdes.go]: Is it better to return different message for 1. len(b) == 0, 2. b[0] is invalid ?

File: util/types/json/serdes.go (hunk 3 above)
Comment (hhkbp2) [inline util/types/json/serdes.go]: The constant 4 is used somewhere else in this file, we could use a const variable for all of them.

**Comment by hhkbp2** [comment]
Rest LGTM

File: util/types/json/serdes.go (hunk 3 above)
Comment (hicqu) [inline util/types/json/serdes.go]: fixed.

**Comment by XuHuaiyu** [comment]
LGTM

File: util/types/json/serdes.go [hunk 6]
```diff
@@ -97,34 +97,39 @@ const (
 	typeCodeLen      int = 1
```
Comment (XuHuaiyu) [inline util/types/json/serdes.go]: comment for these constant?

**Pull request merged**